### PR TITLE
nix/hm-module: init

### PR DESCRIPTION
**Pull request by fufexan**
Automatically start the service along with `graphical-session.target`.

### CODE REVIEW

File: nix/hm-module.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
+self: {
+  config,
+  pkgs,
+  lib,
+  ...
+}: let
+  cfg = config.services.tailray;
+  inherit (lib) mkEnableOption mkPackageOption getExe;
```
Comment (NotAShelf) [inline nix/hm-module.nix]:
```diff
- inherit (lib) mkEnableOption mkPackageOption getExe;
+ inherit (lib.meta) getExe;
+ inherit (lib.options) mkEnableOption mkPackageOption;
```

File: nix/hm-module.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      // {
+        default = self.packages.${pkgs.system}.default;
+      };
+  };
+
+  config = lib.mkIf cfg.enable {
+    home.packages = [cfg.package];
+
+    systemd.user.services.tailray = {
+      Install = {WantedBy = ["graphical-session.target"];};
```
Comment (NotAShelf) [inline nix/hm-module.nix]: Why not `Install.WantedBy`?

**Review by NotAShelf — COMMENTED** [review]
I've been told in the past that the target for the service should be configurable. 

In addition, wouldn't it be better to make tailray service a part of or after tray.target?

File: nix/hm-module.nix (hunk 1 above)
Comment (NotAShelf) [reply nix/hm-module.nix]: I'm pedantic sue me

File: nix/hm-module.nix (hunk 2 above)
Comment (fufexan) [reply nix/hm-module.nix]: Copied from a HM service.

**Review by NotAShelf — APPROVED** [review]

**Pull request merged**